packages/headless/src/controllers/commerce/context/cart/headless-cart.ts

```diff
@@ -170,9 +171,9 @@ export function buildCart(engine: CommerceEngine, props: CartProps = {}): Cart {
       dispatch(setItems([]));
     },
 
-    updateItem(item: CartItem) {
+    async updateItem(item: CartItem) {
       if (isNewQuantityDifferent(item.productId, item.quantity)) {
-        // TODO LENS-1497: log ec.cartAction; if new quantity > previous, 'add', otherwise, 'remove'.
+        await dispatch(logCartAction(item));
```

Comment:
Why await?

Reply:
Sometime the items were updated in the state before the `logCartAction` was triggered. In the IDE, it does mentioned that the `await` statement shouldn't do anything, but when linking to Barca sport there's an effect. Do you have a recommendation to make sure the event is emitted before the state is updated?

Reply:
No await needed anymore :eyes: